rbi/core/array.rbi

```diff
@@ -1037,6 +1037,32 @@ class Array < Object
   sig {returns(T::Enumerator[Elem])}
   def each_index(&blk); end
 
+  # Same as Enumerator#with_index(0), i.e. there is no starting offset.
+  #
+  # If no block is given, a new [`Enumerator`](https://apidock.com/ruby/Enumerator) is
+  # returned that includes the index.
+  #
+  # ```ruby
+  # a = [ "a", "b", "c" ]
+  # a.each_with_index {|x, i| puts "value: #{x}, index: #{i}" }
+  # ```
+  #
+  # produces:
+  #
+  # ```
+  # value: a, index: 0
+  # value: b, index: 1
+  # value: c, index: 2
+  # ```
+  sig do
+    params(
+        blk: T.proc.params(arg0: Elem, arg1: Integer).returns(BasicObject),
+    )
+    .returns(T::Array[Elem])
+  end
+  sig {returns(T::Enumerator[Elem])}
```

Comment:
```suggestion
  sig {returns(T::Enumerator[[Elem,Integer]])}
```

I think that the return type on this overload is incorrect:
```
% irb
irb(main):001:0> [1, 2, 3].each_with_index.to_a
=> [[1, 0], [2, 1], [3, 2]]
```

Reply:
Thank you for reviewing it! 👍